privacyidea/lib/security/aeshsm.py

```diff
@@ -189,9 +186,13 @@ def random(self, length):
                     raise HSMException("Failed to generate random number after multiple retries.")
 
         # convert the array of the random integers to a string
-        return int_list_to_bytestring(r_integers)
+        return bytes(r_integers)
```

Comment:
~~I *think* we might need a case distinction to keep it working on Python 2 here?
https://github.com/LudovicRousseau/PyKCS11/blob/951a9e941cab4f49537e29caf7c9250674e3f4ff/PyKCS11/__init__.py#L1527~~

I just noticed this isn't the Python ``bytes``, but the ``builtins`` ``bytes``, so this may be fine?

Reply:
The problem was that `chr()` returns unicode in Python 3 and thus the `int_list_to_bytestring()` function broke.
Maybe we can use builtins from six, I'll check.

Reply:
:+1: